File: libs/eave-stdlib-py/src/eave/stdlib/core_api/operations/api_documentation_jobs.py
from typing import Unpack, Optional
import uuid
from ... import requests
from eave.stdlib.core_api.models.api_documentation_jobs import (
    ApiDocumentationJob,
    ApiDocumentationJobUpsertInput,
    ApiDocumentationJobListInput,
)
from . import BaseRequestBody, BaseResponseBody, CoreApiEndpoint, CoreApiEndpointConfiguration


# class GetApiDocumentationJobsOperation(CoreApiEndpoint):
#     config = CoreApiEndpointConfiguration(
#         path="/api-documentation-job/query",
#         auth_required=False,
#     )

#     class RequestBody(BaseRequestBody):
#         jobs: Optional[list[ApiDocumentationJobListInput]] = None

#     class ResponseBody(BaseResponseBody):
#         jobs: list[ApiDocumentationJob]

#     @classmethod
#     async def perform(
#         cls,
#         input: RequestBody,
#         team_id: uuid.UUID | str,
#         account_id: Optional[uuid.UUID],
#         access_token: Optional[str],
#         **kwargs: Unpack[requests.CommonRequestArgs],
#     ) -> ResponseBody:
#         response = await requests.make_request(
#             config=cls.config,
#             input=input,
#             team_id=team_id,
#             account_id=account_id,
#             access_token=access_token,
#             **kwargs,
#         )

#         body = await cls.make_response(response, cls.ResponseBody)
#         return body


# class UpsertApiDocumentationJobOperation(CoreApiEndpoint):
#     config = CoreApiEndpointConfiguration(
#         path="/api-documentation-job/upsert",
#         auth_required=False,
#     )

#     class RequestBody(BaseRequestBody):
#         job: ApiDocumentationJobUpsertInput

#     class ResponseBody(BaseResponseBody):
#         job: ApiDocumentationJob

#     @classmethod
#     async def perform(
#         cls,
#         input: RequestBody,
#         team_id: uuid.UUID | str,
#         account_id: Optional[uuid.UUID],
#         access_token: Optional[str],
#         **kwargs: Unpack[requests.CommonRequestArgs],
#     ) -> ResponseBody:
#         response = await requests.make_request(
#             config=cls.config,
#             input=input,
#             team_id=team_id,
#             account_id=account_id,
#             access_token=access_token,
#             **kwargs,
#         )

#         body = await cls.make_response(response, cls.ResponseBody)
#         return body

File: libs/eave-stdlib-ts/src/core-api/operations/api-documentation-jobs.ts
import { makeRequest, RequestArgsTeamId } from "../../requests.js";
import {
  ApiDocumentationJob,
  ApiDocumentationJobListInput,
  ApiDocumentationJobUpsertInput,
} from "../models/api-documentation-jobs.js";
import { CoreApiEndpointConfiguration } from "./shared.js";

export type GetApiDocumentationJobsRequestBody = {
  jobs?: ApiDocumentationJobListInput[];
};

export type GetApiDocumentationJobsResponseBody = {
  jobs: ApiDocumentationJob[];
};

export class GetApiDocumentationJobsOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/api-documentation-job/query",
  });
  /**
   * Performs an asynchronous request with the provided arguments.
   *
   * @param args - An object that contains the team ID and the input for the API documentation jobs request.
   * @returns A promise that resolves to the response body of the API documentation jobs request.
   */
  static async perform(
    args: RequestArgsTeamId & { input: GetApiDocumentationJobsRequestBody },
  ): Promise<GetApiDocumentationJobsResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <GetApiDocumentationJobsResponseBody>await resp.json();
    return responseData;
  }
}

export type UpsertApiDocumentationJobRequestBody = {
  job: ApiDocumentationJobUpsertInput;
};

export type UpsertApiDocumentationJobResponseBody = {
  job: ApiDocumentationJob;
};

export class UpsertApiDocumentationJobOperation {
  static config = new CoreApiEndpointConfiguration({
    path: "/api-documentation-job/upsert",
  });
  /**
   * Performs an asynchronous operation to upsert API documentation job.
   *
   * @param args - An object that contains the team ID and the input for the upsert API documentation job request.
   * @returns A promise that resolves to the response body of the upsert API documentation job.
   */
  static async perform(
    args: RequestArgsTeamId & { input: UpsertApiDocumentationJobRequestBody },
  ): Promise<UpsertApiDocumentationJobResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <UpsertApiDocumentationJobResponseBody>(
      await resp.json()
    );
    return responseData;
  }
}
